### PR TITLE
Fix potential null exception of OverlayEntry in Slider

### PR DESCRIPTION
**Pull request by liumcse**
When `Slider` is disposed while user is interacting with the Slider, null pointer exception will be thrown. This PR fixes it by saving `overlayEntry` to a local variable when calling `showValueIndicator()` instead of using `!` to force converting potentially null value to non-null value.

Also using `?` instead of `!` In `dispose()` method.

*List which issues are fixed by this PR. You must list at least one issue.*

Internal bug: b/273666179, b/192329942

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/slider.dart [hunk 1]
```diff
@@ -1064,7 +1063,7 @@ class _RenderSlider extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
       curve: Curves.fastOutSlowIn,
     )..addStatusListener((AnimationStatus status) {
       if (status == AnimationStatus.dismissed && _state.overlayEntry != null) {
-        _state.overlayEntry!.remove();
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/slider.dart:1067]: The original code seems fine to me, I think if this become null somehow. Some thing may be broken and we would want to know

File: packages/flutter/lib/src/material/slider.dart [hunk 2]
```diff
@@ -908,7 +906,7 @@ class _SliderState extends State<Slider> with TickerProviderStateMixin {
 
   void showValueIndicator() {
     if (overlayEntry == null) {
-      overlayEntry = OverlayEntry(
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/slider.dart:911]: I can't see how it is possible that overlayEntry become null before you insert.

File: packages/flutter/test/material/slider_test.dart [hunk 3]
```diff
@@ -352,6 +352,45 @@ void main() {
     expect(SchedulerBinding.instance.transientCallbackCount, equals(0));
   });
 
+  testWidgets('Should not throw exception when dragged after Slider is disposed', (WidgetTester tester) async {
```
Comment (chunhtai) [inline packages/flutter/test/material/slider_test.dart:355]: This test pass even without the change, is this testing a real issue?

**Review by chunhtai — COMMENTED** [review]
I can't think of a way the showValueIndicator can be null, I think there may be something more going on there

**Comment by TahaTesser** [comment]
Thanks for updating the PR. I'll take a look on Monday.

File: packages/flutter/test/material/slider_test.dart (hunk 3 above)
Comment (TahaTesser) [reply packages/flutter/test/material/slider_test.dart:355]: @liumcse 
It seems that the test passes without the fix on the master.

**Comment by TahaTesser** [comment]
cc: @chunhtai
In order to address this we'd need the internal bug to be tracked in a Flutter issue with more details.
We can probably close this in the meantime since the fix and test aren't clear.

**Comment by chunhtai** [comment]
I agree. @liumcse I think we need a repro before we can fix this issue. Otherwise, it would likely to be a workaround that would bite us back in the future. I will be closing this PR, feel free to reopen if you have time to come back to this pr

**Pull request closed without merging**